core_v14/src/value/mod.rs

```diff
@@ -34,7 +34,7 @@ use std::fmt::Debug;
 /// sequence, array and composite types can all be represented with [`Composite`]. Only enough information
 /// is preserved here to construct a valid value for any type that we know about, and be able to verify
 /// that a given value is compatible with some type (see the [`scale_info`] crate), if we have both.
-#[derive(Clone, PartialEq)]
```

Comment:
I guess these are serialized to be stored in the database?

My only hesitation here is that the serialize and deserialize impls aren't symmetric. I wonder whether it would be better to write serialize impls that transform like so:

`Value::Composite(Composite::Unnamed(vec![values]))` => sequence of values.
`Value::Composite(Composite::Named(vec![keyvalues]))` => map of key+value pairs.
`Value::Variant(..)` => serde data model variant.
`Value::Primitive(Primitive::Foo(..)) => serialize straight to the primitive value in question.
`Value::BitSequence(..)` => serialize straight to the bitsequence?

This would mirror Deserialize more closely, which attempt to deserialize based into the relevant structure based on the type of Value (whereas by default it would always try to deserialize into an enum variant containing eg a composite which is another enum variant and so on).

What do you think?

Reply:
If you like, I can work on said impls this afternoon!

Reply:
PR here: https://github.com/paritytech/desub/pull/69